### PR TITLE
Better instance shortcuts on macOS

### PR DESCRIPTION
**Pull request by iivusly**
## When creating shortcuts on macOS, a bash script is generated to the desktop.
Realistically, we don't want to do that. We can't find the shortcut through Spotlight, the Dock, or through Launchpad. That's why I created a simple fix, by wrapping the generated script in an Application bundle, so we can treat it as if it was a separate Application.

### CODE REVIEW

File: launcher/FileSystem.cpp [hunk 1]
```diff
@@ -770,9 +770,37 @@ QString getDesktopDir()
 bool createShortcut(QString destination, QString target, QStringList args, QString name, QString icon)
 {
 #if defined(Q_OS_MACOS)
-    destination += ".command";
+    // Create the Application
+    QDir applicationDirectory = QStandardPaths::writableLocation(QStandardPaths::ApplicationsLocation) + "/Prism Launchers/";
```
Comment (DioEgizio) [inline launcher/FileSystem.cpp]: haven't tested this yet, but does this return $HOME/Applications/ or /Applications/ ?

Comment (iivusly) [reply launcher/FileSystem.cpp]: When I have tested it, it returns $HOME/Applications

**Review by DioEgizio — APPROVED** [review]
nice qol change! Though doesn't work because https://github.com/PrismLauncher/PrismLauncher/issues/1229, but that can be fixed in a different pr

File: launcher/FileSystem.cpp [hunk 2]
```diff
@@ -770,9 +770,37 @@ QString getDesktopDir()
...
 
-    QFile f(destination);
+    if (!applicationDirectory.exists())
+        applicationDirectory.mkpath(".");
```
Comment (TheKodeToad) [inline launcher/FileSystem.cpp]:
```suggestion
    if (!applicationDirectory.mkpath(".")) {
        qWarning() << "Couldn't create application directory";
        return false;
    }
```

File: launcher/FileSystem.cpp [hunk 3]
```diff
@@ -770,9 +770,37 @@ QString getDesktopDir()
...
+    application.mkpath(".");
+
+    QDir content = application.path() + "/Contents/";
+    QDir resources = content.path() + "/Resources/";
+    QDir binaryDir = content.path() + "/MacOS/";
+    QFile info = content.path() + "/Info.plist";
+
+    content.mkpath(".");
+    resources.mkpath(".");
+    binaryDir.mkpath(".");
```
Comment (TheKodeToad) [inline launcher/FileSystem.cpp]:
```suggestion
    if (!(content.mkpath(".") && resources.mkpath(".") && binaryDir.mkpath("."))) {
        qWarning() << "Couldn't create directories within application";
        return false;
    }
```

File: launcher/FileSystem.cpp [hunk 4]
```diff
@@ -770,9 +770,37 @@ QString getDesktopDir()
...
+        applicationDirectory.mkpath(".");
+
+    QDir application = applicationDirectory.path() + "/" + name + ".app/";
+
+    if (application.exists()) {
+        qWarning() << "Application already exists!";
+        return false;
+    }
+
+    application.mkpath(".");
```
Comment (TheKodeToad) [inline launcher/FileSystem.cpp]:
```suggestion
    if (!application.mkpath(".")) {
        qWarning() << "Couldn't create application";
        return false;
    }
```

File: launcher/FileSystem.cpp (hunk 1 above)
Comment (TheKodeToad) [inline launcher/FileSystem.cpp]:
```suggestion
    // Create the Application
    QDir applicationDirectory = QStandardPaths::writableLocation(QStandardPaths::ApplicationsLocation) + "/Prism Launcher Instances/";
```

or something maybe?

**Review by TheKodeToad — APPROVED** [review]

File: launcher/FileSystem.cpp (hunk 1 above)
Comment (DioEgizio) [reply launcher/FileSystem.cpp]: Maybe also prism launcher should be rather LAUNCHER_DISPLAYNAME

**Review by TheKodeToad — APPROVED** [review]

**Comment by Ryex** [comment]
#1229 now should be fixed. do we want to merge this for 7.2 after all?

**Review by Ryex — APPROVED** [review]
Looks good.

**Comment by DioEgizio** [comment]
I mean it's quite nice QoL. though it needs a rebase because conflicts

**Comment by Ryex** [comment]
and now it doesn't build :P

**Comment by iivusly** [comment]
Everything should be good now, accidentally made dup. variables because of 3 am stupidity lol

**Pull request merged**